src/sagas/trendingGifs.js

```diff
@@ -0,0 +1,84 @@
+import { 
+  call,
+  takeEvery,
+  put,
+  select
+} from 'redux-saga/effects';
+import trendingGifs from '../actions/trendingGifs';
+import giftService from '../giphy_sdk/gif';
+import { getFavoriteGifs } from '../storage/favoriteGifs';
+
+const { types } = trendingGifs;
+
+const selectData2Send = (state) => {
+  return {offset: state.trendingGifs.pagination.offset, q: state.trendingGifs.query};
+};
+
+const selectSelectedID = (state) => {
+  return state.trendingGifs.selectedID;
+};
+
+const injectFavoriteMarked = (dataGifs, total_count=0) => {
+  total_count = total_count === 0 ? 50000 : total_count;
+
+  return new Promise((resolve) => {
+    getFavoriteGifs({count: total_count})
+      .then(({data: favorites}) => {
+        const existInFav = (gif) => favorites.some(f => f.id === gif.id);
+        dataGifs.forEach(gif => {
+          if (existInFav(gif)) {
+            gif.isFavorite = true;
+          }
+        });
+        resolve(dataGifs);
+      });
+  });
+};
+
+export function* loadTrendingGifsSaga() {
+  const { creators } = trendingGifs;
+  // console.log('[SAGA] loadTrendingGifsSaga...');
+  try {
+    const data2Send = yield select(selectData2Send);
+    let trendingGifsResponse = {};
+    // console.log('​exportfunction*loadTrendingGifsSaga -> offset', offset);
+    if (data2Send.q === '') {
+      delete(data2Send.q);
+      trendingGifsResponse = yield call(giftService.trending, data2Send );
+    } else {
+      trendingGifsResponse = yield call(giftService.search, data2Send );
+    }
+
+    yield call(
+      injectFavoriteMarked,
```

Comment:
Since this loading is a side effect, this actually a good place to execute it (inside a Saga), nice! 👍